sql-plugin/src/main/spark330db/scala/org/apache/spark/sql/rapids/execution/GpuExecutorBroadcastHelper.scala

```diff
@@ -65,11 +66,23 @@ object GpuExecutorBroadcastHelper {
     // Use the GPU Shuffle Coalesce iterator to concatenate and load batches onto the 
     // host as needed. Since we don't have GpuShuffleCoalesceExec in the plan for the 
     // executor broadcast scenario, we have to use that logic here to efficiently 
-    // grab and release the semaphore while doing I/O
+    // grab and release the semaphore while doing I/O. We wrap this with GpuCoalesceIterator
+    // to ensure this always a single batch for the following step.
     val iter = shuffleDataIterator(shuffleData)
-    new GpuShuffleCoalesceIterator(
-      new HostShuffleCoalesceIterator(iter, targetSize, metricsMap),
-      dataTypes, metricsMap).asInstanceOf[Iterator[ColumnarBatch]]
+    new GpuCoalesceIterator(
+      new GpuShuffleCoalesceIterator(
+        new HostShuffleCoalesceIterator(iter, targetSize, metricsMap),
+        dataTypes, metricsMap).asInstanceOf[Iterator[ColumnarBatch]],
+      dataTypes,
+      RequireSingleBatch,
+      metricsMap(NUM_INPUT_ROWS), // numInputRows
+      metricsMap(NUM_INPUT_BATCHES), // numInputBatches
+      metricsMap(NUM_OUTPUT_ROWS), // numOutputRows
+      metricsMap(NUM_OUTPUT_BATCHES), // numOutputBatches
+      metricsMap(COLLECT_TIME), // collectTime
+      metricsMap(CONCAT_TIME), // concatTime
+      metricsMap(OP_TIME), // opTime
```

Comment:
I think this is going to end up over counting many of these metrics, as we pass the metrics map to the HostShuffleCoalesceIterator as well as GpuShuffleCoalesceIterator.  If all of these things end up incrementing input rows, output rows, etc, then we're going to get numbers way off from what one would expect to be reported for a broadcast join.  I'm wondering if we were already over counting metrics from before since we're passing the metrics map to GpuShuffleCoalesceIterator and HostShuffleCoalesceIterator.  I suspect we only want to pass a map containing OP_TIME and CONCAT_TIME to these iterators and nothing else.  For GpuCoalesceIterator which takes individual metrics, we can pass NoopMetric for the ones we don't want to count.